### ISSUE TITLE
Bug Report: lag spike in week 3

### ISSUE THREAD

**Issue by NiftyAC** [FunkinCrew/Funkin#3754]
### Issue Checklist

- [x] I have properly named my issue
- [x] I have checked the Issues/Discussions pages to see if my issue has already been reported

### Platform

Newgrounds (Web/HTML5)

### Browser

None

### Version

0.5.1, 0.5.2, 0.5.3

### Description (include any images, videos, errors, or crash logs)

I don't know what causes it and I tried without mods and it still happens. Randomly when playing week 3 in 0.5.2 and 0.5.3 the game has a major lag spike that makes me miss a few notes, it usually occurs at the fastest part.

**Comment by NotHyper-474** [comment]
Seems related to https://github.com/FunkinCrew/Funkin/issues/3495 and https://github.com/FunkinCrew/Funkin/issues/3667

**Comment by Hundrec** [comment]
Are these lag spikes causing notes to briefly halt?
Do they still happen if you set your input offsets to 0ms (if they aren't already)?

**Comment by NiftyAC** [comment]
> Are these lag spikes causing notes to briefly halt?
> 
> Do they still happen if you set your input offsets to 0ms (if they aren't already)?

The entire game freezes for half a second to a whole second and makes it unplayable.

**Comment by Hundrec** [comment]
Sounds like the issue Hyper referenced above.
Can you provide information regarding your iinput offsets and provide a video of the lag spike?

**Comment by NiftyAC** [comment]
> Sounds like the issue Hyper referenced above.
> 
> Can you provide information regarding your iinput offsets and provide a video of the lag spike?

It will have to be later but yeah. Also, I haven't changed any settings except for enabled down scroll and have 165 fps. I don't know if that would change anything.

**Comment by NiftyAC** [comment]
> > Sounds like the issue Hyper referenced above.
> > 
> > Can you provide information regarding your iinput offsets and provide a video of the lag spike?
> 
> It will have to be later but yeah. Also, I haven't changed any settings except for enabled down scroll and have 165 fps. I don't know if that would change anything.

https://github.com/user-attachments/assets/0d7492c5-3b0f-4460-9442-79c123237a22

There's the video.
I don't know what to do for the input offsets and how to see those. All I see is visual offset which is 0 but everything is fine with offsets and I have no problems there so I'd say they're the default ones.

**Comment by Hundrec** [comment]
Thanks for the video!
Have you ever encountered these lag spikes on an unmodified build of 0.5.3?

**Comment by NiftyAC** [comment]
> Thanks for the video!
> Have you ever encountered these lag spikes on an unmodified build of 0.5.3?

Yes. I forgot to disable the mods to take the video but when I disable them it still happens. I don't know if this has anything to do with it but it usually happens in that same area in the song when the train goes past but only that specific section. Any other time the train passes it works smooth.

**Comment by Hundrec** [comment]
Oh! That's good to know.
Then the lag spikes are occurring because your game is loading the train for the first time. I'm not sure I've ever encountered this before. 
How is your computer's performance?

**Comment by NiftyAC** [comment]
> Oh! That's good to know.
> Then the lag spikes are occurring because your game is loading the train for the first time. I'm not sure I've ever encountered this before. 
> How is your computer's performance?

I have a pretty high power PC so I know it isn't that, I run everything pretty well. It also isn't the first time the train goes past and I can link me playing the entire song if you'd like. I can record that just so you can see all of it.

**Comment by Hundrec** [comment]
Oh, I misinterpreted your explanation!

If the location of the lag spike is consistent within the song(s), use the chart editor to try to identify around which section this happens so I can get a better idea of what could be happening there.

**Comment by NiftyAC** [comment]
> Oh, I misinterpreted your explanation!
> 
> If the location of the lag spike is consistent within the song(s), use the chart editor to try to identify around which section this happens so I can get a better idea of what could be happening there.

It's at about 1:07 into the song. A minute and seven seconds was the place that it consistently happened.

**Comment by Hundrec** [comment]
Just checked out that region of Pico Erect and didn't find anything besides a few camera zooms. Have you tried playing the song on the Itch.io / Newgrounds HTML5 versions?

**Comment by NiftyAC** [comment]
> Just checked out that region of Pico Erect and didn't find anything besides a few camera zooms. Have you tried playing the song on the Itch.io / Newgrounds HTML5 versions?
> 
> 

I didn't know the newgrounds one was updated for the newest version. I'll try it when I can but I've had problems with the pico erect on 0.5.2 and the new 0.5.3 so it's weird if it's my pc. The 0.5.0 and earlier I didn't have problems with but I don't think I really played 0.5.1 (this was confusing I know)

**Comment by Hundrec** [comment]
No worries, we've had a confusing round of patches since 0.5.0
Take your time and let me know what you find when you're ready

**Comment by NiftyAC** [comment]
> No worries, we've had a confusing round of patches since 0.5.0
> 
> Take your time and let me know what you find when you're ready 

I think that when I can tomorrow I'm gonna play through all the versions I have which is 0.4.x since I don't exactly remember to 0.5.3 and see which ones have that glitch. I can't get the P rank on pico nightmare because of the glitch but I could on older versions and got it on every song so it's really weird.

**Comment by Hundrec** [comment]
Heads up: Be careful of the save file wiping bug! 
Remember to back up your **Funkin1.sol** file before attempting to play older versions, and **do not** open 0.5.2 to be safe.

**Comment by NiftyAC** [comment]
> Heads up: Be careful of the save file wiping bug! 
> 
> Remember to back up your **Funkin1.sol** file before attempting to play older versions, and **do not** open 0.5.2 to be safe. 

So THATS what happened to my save. I had like 8 songs beaten and it opened to saying it couldn't load my save. It wasn't an important save and it's already gone so I should be fine. I'll just have to fix my settings again.

**Comment by Hundrec** [comment]
If you got a popup warning you that the save couldn't load, your old save has been moved to Funkin1000.sol in the same %appdata%\FunkinCrew directory as your new save.

**Comment by NiftyAC** [comment]
> If you got a popup warning you that the save couldn't load, your old save has been moved to Funkin1000.sol in the same %appdata%\FunkinCrew directory as your new save.

I have a small question related to saves, is mine messed up or does progress not save from some versions? I had a 100% complete save file before 0.5.0 and it was all gone on the pico songs update

**Comment by Hundrec** [comment]
Ah, a few other people have reported losing their save files before 0.5.2's save bug:
https://github.com/FunkinCrew/Funkin/issues/3369
https://github.com/FunkinCrew/Funkin/issues/3524

As far as I know, the cause of these pre-0.5.2 save wipes is still unknown.

However, you should be able to retrieve your save any time you see that popup warning.

**Comment by NiftyAC** [comment]
> Ah, a few other people have reported losing their save files before 0.5.2's save bug:
> 
> https://github.com/FunkinCrew/Funkin/issues/3369
> 
> https://github.com/FunkinCrew/Funkin/issues/3524
> 
> 
> 
> As far as I know, the cause of these pre-0.5.2 save wipes is still unknown.
> 
> 
> 
> However, you should be able to retrieve your save any time you see that popup warning.

Okay, thank you for telling me because I thought it was all gone. I have no problem with replaying it all but I had all perfect scores so it was a little annoying when the next update it all just disappeared. But I'll try all the versions tomorrow to see if the lagspike only happens after the pico update or anything. I know game development it pretty difficult so it happens.

**Comment by NiftyAC** [comment]
> Ah, a few other people have reported losing their save files before 0.5.2's save bug: #3369 #3524
> 
> As far as I know, the cause of these pre-0.5.2 save wipes is still unknown.
> 
> However, you should be able to retrieve your save any time you see that popup warning.

the glitch of the lagspike started on version 0.5.1 and happen on 0.5.2 and 0.5.3 too and there's a few minor lagspikes that arent bad in it at random times

**Comment by NiftyAC** [comment]
> Ah, a few other people have reported losing their save files before 0.5.2's save bug:
> 
> https://github.com/FunkinCrew/Funkin/issues/3369
> 
> https://github.com/FunkinCrew/Funkin/issues/3524
> 
> 
> 
> As far as I know, the cause of these pre-0.5.2 save wipes is still unknown.
> 
> 
> 
> However, you should be able to retrieve your save any time you see that popup warning.

This didn't happen before but with no mods the lagspike happens in every song. I played all the way through 0.5.3 before I believe with no problems and and it happens on every song instead of just pico. This glitch is probably already listed but I thought I'd put this here in case it isn't.

**Comment by Hundrec** [comment]
Thanks for the information. We'll see how the devs tackle the many lag spikes and performance issues that have been cropping up since 0.5.0

**Comment by Hundrec** [comment]
Does this still happen in 0.6.3?

**Comment by NiftyAC** [comment]
> Does this still happen in 0.6.3?

No it is fixed in the new version.
